### PR TITLE
BIP 175: Pay to Contract Protocol

### PR DESCRIPTION
**Pull request by omarshibli**
This proposal was submitted to the [mailing list](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2017-September/014896.html) on August 14th, we got feedback and addressed the concerns, I am now requesting a BIP number.

### CODE REVIEW

**Comment by christianrolandso** [comment]
Zzzzzzz

File: bip-draft.mediawiki [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+  Author: Omar Shibli <omar@commerceblock.com>
+          Nicholas Gregory <nicholas@commerceblock.com>
+  Comments-Summary: No comments yet.
+  Comments-URI: https://github.com/bitcoin/bips/wiki/Comments:BIP-0999
+  Status: Draft
+  Type: Informational Track
+  Created: 2017-07-17
+</pre>
+
+&ast; 999 is a temporary BIP number.
```
Comment (luke-jr) [inline bip-draft.mediawiki]: Please don't self-assign numbers, even temporarily.

File: bip-draft.mediawiki [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+==Reference==
+
+* [[bip-0032.mediawiki|BIP32 - Hierarchical Deterministic Wallets]]
+* [[bip-0043.mediawiki|BIP43 - Purpose Field for Deterministic Wallets]]
+* [[bip-0044.mediawiki|BIP44 - Multi-Account Hierarchy for Deterministic Wallets]]
+* [[https://arxiv.org/abs/1212.3257|Homomorphic Payment Addresses and the Pay-to-Contract Protocol]]
+
+==Copyright==
+
+This document is placed in the public domain.
```
Comment (luke-jr) [inline bip-draft.mediawiki]: See BIP 2 for a list of acceptable licenses.

**Comment by omarshibli** [comment]
hey @luke-jr, we are addressed your points, please review.

**Comment by luke-jr** [comment]
By compatibility, I meant more of (whatever the particular case may be; these are examples):

* "contracts can be verified separately from the sender's wallet software, and then a normal address is used"
* "This is incompatible with multisig P2SH wallets"
* "The wallet is assumed to use BIP 43 derivation"

Note that BIP 43 (and even BIP 32 / HD wallets in general) is *not* universal, and your proposal's connection to them seems unnecessary and pointless. What is the benefit of even doing anything HD-based, since the end key requires non-deterministic metadata (contracts) to find?

Anyhow, let's use BIP **175** for this.

**Comment by omarshibli** [comment]
Got it, we will work on clearing this out.

**Comment by omarshibli** [comment]
Thanks for your feedback @luke-jr , you are raising a valid point regarding the backward compatibility with BIP32 and other related BIPs, we have updated the BIP accordingly.
Regarding your question, I think we need to work better on the motivation section here, but I still see benefit here in using BIP32 derivation mechanism or something similar to derive non-enumerable set as well.
I think the main benefit is that ,as you mentioned, BIP32 is not universal in a sense that it only covers use cases where the derivation paths set is an enumerable, yet in our use case we need derive wallets from non-enumerable set and a single seed.

Much appreciated.

**Comment by luke-jr** [comment]
When you're ready, don't forget to rename the file and add to README

File: bip-0175.mediawiki [hunk 3]
```diff
@@ -0,0 +1,252 @@
+<pre>
+  BIP: 175
+  Layer: Applications
+  Title: Pay to Contract Protocol
+  Author: Omar Shibli <omar@commerceblock.com>
+          Nicholas Gregory <nicholas@commerceblock.com>
+  Comments-Summary: No comments yet.
+  Comments-URI: https://github.com/bitcoin/bips/wiki/Comments:BIP-0175
+  Status: Draft
+  Type: Informational Track
```
Comment (luke-jr) [inline bip-0175.mediawiki]: drop "Track"

Comment (omarshibli) [reply bip-0175.mediawiki]: done

**Pull request merged**